apps/src/applab/commands.js

```diff
@@ -1790,15 +1791,67 @@ applabCommands.handleReadValue = function(opts, value) {
 };
 
 applabCommands.getList = function(opts) {
+  validateGetListArgs(opts.tableName, opts.columnName);
   var onSuccess = handleGetListSync.bind(this, opts);
   var onError = handleGetListSyncError.bind(this, opts);
   Applab.storage.readRecords(opts.tableName, {}, onSuccess, onError);
 };
 
+var validateGetListArgs = function(tableName, columnName) {
+  const datasetNames = datasetLibrary.datasets.map(d => d.name);
+  if (datasetNames.indexOf(tableName) === -1) {
+    outputWarning(
+      tableName +
+        ' is not a data set in this project.' +
+        ' Check the Data tab to see the names of your tables.'
+    );
+  }
+  for (let dataset of datasetLibrary.datasets) {
```

Comment:
Nit: It seems like we are using this 'for-loop' method of looking up the specific data-set multiple times. Is there a different way to look the requested dataset when we know the name of the requested data-set?